From 8a5de0b6cf1090d7f29f3974ec79c32776cf2745 Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jami.kettunen@protonmail.com>
Date: Tue, 31 Aug 2021 00:15:58 +0300
Subject: [PATCH] freedreno/pm4: Use unsigned instead of uint to fix musl build

Upstream-Status: Backport

Fixes the following error I noticed when building against aarch64 with
musl libc:

  In file included from ../src/freedreno/decode/crashdec.h:38,
                   from ../src/freedreno/decode/crashdec.c:40:
  ../src/freedreno/common/freedreno_pm4.h:104:15: error: unknown type name 'uint'
    104 | static inline uint
        |               ^~~~
  ../src/freedreno/common/freedreno_pm4.h:105:25: error: unknown type name 'uint'; did you mean 'int'?
    105 | pm4_calc_odd_parity_bit(uint val)
        |                         ^~~~
        |                         int

Signed-off-by: Jami Kettunen <jami.kettunen@protonmail.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/19665>
---
 src/freedreno/common/freedreno_pm4.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/freedreno/common/freedreno_pm4.h b/src/freedreno/common/freedreno_pm4.h
index 8f958953d693..091247e709a0 100644
--- a/src/freedreno/common/freedreno_pm4.h
+++ b/src/freedreno/common/freedreno_pm4.h
@@ -105,8 +105,8 @@ pm4_pkt7_hdr(uint8_t opcode, uint16_t cnt)
 #define cp_type3_opcode(pkt) (((pkt) >> 8) & 0xFF)
 #define type3_pkt_size(pkt)  ((((pkt) >> 16) & 0x3FFF) + 1)
 
-static inline uint
-pm4_calc_odd_parity_bit(uint val)
+static inline unsigned
+pm4_calc_odd_parity_bit(unsigned val)
 {
    return (0x9669 >> (0xf & ((val) ^ ((val) >> 4) ^ ((val) >> 8) ^
                              ((val) >> 12) ^ ((val) >> 16) ^ ((val) >> 20) ^
-- 
2.39.2

